lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx

```diff
@@ -558,15 +564,37 @@ Scalar GeneralLinearModelAlgorithm::maximizeReducedLogLikelihood()
     return initialReducedLogLikelihood;
   }
   // At this point we have an optimization problem to solve
-  // Define the optimization problem
+  // Define the optimization problem. It is passed to solver_ attribute
+  // in order to have access to the locally built problem outside of
+  // the method, but it will not be the actual solver used for the
+  // optimization as it may be replaced by a multi-start version
   OptimizationProblem problem(reducedLogLikelihoodFunction);
   problem.setMinimization(false);
   problem.setBounds(optimizationBounds_);
   solver_.setStartingPoint(initialParameters);
   solver_.setProblem(problem);
-  LOGINFO(OSS(false) << "Solve problem=" << problem << " using solver=" << solver_);
-  solver_.run();
-  const OptimizationAlgorithm::Result result(solver_.getResult());
+  OptimizationAlgorithm actualSolver(solver_);
```

Comment:
That seems to be a poor fix for a real problem. I'd rather opt for Joseph's solution.